### ISSUE TITLE
Failed update to Blazor 0.4.0

### ISSUE THREAD

**Issue by simonziegler** [aspnet/Blazor#1041]
Hi, I'm testing a system with Blazor 0.3.0 and am keen to pick up some of the new stuff in 0.4.0. However when I update and try running (client/server), I get presented with the following in my browser. Any ideas what may be the issue?

Thanks!

![image](https://user-images.githubusercontent.com/11708435/41867493-30bce592-78ab-11e8-9e0b-0ac14cb0c2ca.png)

**Comment by Andrzej-W** [comment]
Try to create a new 0.4.0 Blazor project in VS and then compare content of all *.csproj files to project files in your "old" project. Also check global.json - maybe you have a reference to old sdk version.

**Comment by chucker** [comment]
OK, next up, check that your `csproj` file doesn't have any stray tags. There probably shouldn't be any `ItemGroup`s with `Add` or `Remove` directives for files. See the [0.4 release notes](https://blogs.msdn.microsoft.com/webdev/2018/06/07/blazor-0-4-0-experimental-release-now-available/) for an example project file.

**Comment by simonziegler** [comment]
Thanks @chucker and @Andrzej-W. I'll give this a go later on and report back. Much appreciated!

**Comment by simonziegler** [comment]
Thanks for the suggestions. There were a couple of anomalies that I fixed (global.json has a preview reference in particular), but the issue is not resolved. I'll start digging into what happens as the server starts up, as I suspect it is crashing at some point. Any further ideas would be much appreciated.

Kind regards,     Simon

**Comment by Andrzej-W** [comment]
Although you cannot debug Blazor code in VS you can debug server code (it is standard ASP.NET Core project). Set server project as "startup project" in solution explorer, set breakpoint on Main method in Program class, press F5 and pray :)
You mentioned that you have VS 2017 15.7.1. I have used Blazor 0.4.0 in VS 2017 15.7.3, and now I have just upgraded to 15.7.4. I don't know if that matters.

**Comment by simonziegler** [comment]
Thanks. I already tried that and there’s a crash before the first line of the Server’s Main function (or it skips over my break point). I need to spend a bit of time working out what is what here.

**Comment by simonziegler** [comment]
I mostly have a fix now. To help anybody else who is experiencing this problem, here is what I found and the fix that gets me (almost working). I then explain the residual problem.

What was catching me was that Visual Studio and its SDKs updated in the last few days, and that these updates were needed for Blazor 0.4.0. I had been using the VS Preview, which now no longer needs to be the case. I had to update the Dot Net to version 2.1.1, requiring the SDK 2.1.301 (https://www.microsoft.com/net/download/windows). My app now starts successfully.

HOWEVER, I'm not out of the woods. My Client crashes on its first HttpPost, which is for a user log in. While the Client crashes, I am still able to put a breakpoint in the server code and trace its successful continued operation after the Client crash. I get the following in the browser's console explaining what has gone wrong in blazor.js. Any ideas please?


![image](https://user-images.githubusercontent.com/11708435/41904503-7bc73b8e-7930-11e8-98a8-694798420066.png)

Kind regards,    Simon

**Comment by Suchiman** [comment]
Is there anything immediately above the Exception? A warning from the browser? `HttpClient` calls `fetch`  and in this case, fetch returned `TypeError: NetworkError when attempting to fetch resource` which is passed through into .NET and goes unhandled. Is there anything interesting on the Network tab of the developer tools?

**Comment by simonziegler** [comment]
I tried something radical that failed. I created an entirely new Blazor 0.4.0 hosted app,  which naturally worked with the weather fetching code. I then copied all of my Blazor 0.3.0 app's code over. This way I could tell that I was not making some csproj error. The new app has exactly the same failure as the old app.

I believe that the issue may be due to the 0.4.0 camelCase modification, or something similar. I note from the browser console output that the error occurs in the SendJsonAsync method, which had its deserialization function call changed recently.

Does anybody have a view as to whether I might be right here, and either way how I can resolve this issue?

Many thanks,      Simon
